src/main/controllers/helpers/CitizenHubHelper.ts

```diff
@@ -127,13 +127,80 @@ export const shouldHubLinkBeClickable = (status: HubLinkStatus, linkName: string
     return false;
   }
 
-  if (status === HubLinkStatus.WAITING_FOR_TRIBUNAL && linkName !== HubLinkNames.RespondentApplications) {
+  if (
+    status === HubLinkStatus.WAITING_FOR_TRIBUNAL &&
+    linkName !== HubLinkNames.RespondentApplications &&
+    linkName !== HubLinkNames.RequestsAndApplications
+  ) {
     return false;
   }
 
   return true;
 };
 
+export const getAllClaimantApplications = (userCase: CaseWithId): GenericTseApplicationTypeItem[] => {
+  return userCase.genericTseApplicationCollection?.filter(item => item.value.applicant === Applicant.CLAIMANT);
+};
+
+export const updateYourApplicationsStatusTag = (
+  allClaimantApplications: GenericTseApplicationTypeItem[],
+  userCase: CaseWithId
+): void => {
+  const claimantAppsWaitingForTribunal = allClaimantApplications.filter(
+    it => it.value.applicationState === HubLinkStatus.WAITING_FOR_TRIBUNAL
+  );
+
+  let citizenHubHighestPriorityStatus: HubLinkStatus | undefined;
+
+  claimantAppsWaitingForTribunal.forEach(claimantApp => {
+    const respondCollection = claimantApp.value?.respondCollection;
+
+    if (!respondCollection || respondCollection.length <= 1) {
```

Comment:
I'm not quite following what's happing in this section down to line 182. Could you please annotate with some comments to explain each section.
I also think more test cases may be required to get full coverage for these conditions

Reply:
Added comments